python/nammu/utils/__init__.py

```diff
@@ -24,14 +24,59 @@
 import logging
 import re
 import urllib
+from UserDict import UserDict
 from java.lang import ClassLoader, System
 from java.awt import Font
+from javax.swing import JFrame, JOptionPane
 
 '''
 This is a compilation of methods to be used from all Nammu classes.
 '''
 
 
+class NammuException(Exception):
+    pass
+
+
+class ConfigDict(UserDict):
+    '''
+    Thin wrapper around a dictionary class.  It has only an additional
+    attribute holding the path to the configuration file.
+    '''
+    def __init__(self, config_file, *args):
+        '''
+        Constructor of the `ConfigDict` class.  `config_file` is the path to
+        the configuration file, the rest of the arguments are passed to the
+        constructor of the base class.
+        '''
+        self.config_file = config_file
+        UserDict.__init__(self, *args)
+
+    def __getitem__(self, key):
+        '''
+        Return the value corresponding to the `key` in the dictionary, if the
+        key is available.  If it is not, show an error dialog and close Nammu.
+        '''
+        if key in self:
+            return UserDict.__getitem__(self, key)
+        else:
+            # We instantiate `JOptionPane()` so that we can monkeypatch
+            # `showMessageDialog` in the tests.
+            pane = JOptionPane()
+            pane.showMessageDialog(JFrame().getContentPane(),
+                                   "The configuration file \"%s\"\n"
+                                   "is corrupted (missing \"%s\" key).\n\n"
+                                   "Please delete the file, or "
+                                   "rename it if you want to keep a "
+                                   "backup copy,\nthen restart Nammu.\n\n"
+                                   "If you need help, open an issue in "
+                                   "https://github.com/oracc/nammu." %
```

Comment:
Is there space on the line to link to the [new issue page](https://github.com/oracc/nammu/issues/new) directly?

Reply:
The links are't clickable and can't be even copied, I'm not sure it's worth putting a long URL that the users would have to manually type themselves

Reply:
Anyway, it would look like this:
![image](https://user-images.githubusercontent.com/765740/69164910-aed35f80-0ae8-11ea-98bb-450f870a862e.png)
Not exceedingly long

Reply:
Fair point, can leave it as it is.